catalog/models.py

```diff
@@ -1,3 +1,124 @@
+
+class Genre(models.Model):
+    """Model representing a book genre."""
+
+    name = models.CharField(max_length=200, help_text='Enter a book genre (e.g. Science Fiction)')
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return self.name
+
+class Language(models.Model):
+    """Model representing the language for a book."""
+
+    language = models.CharField(max_length=100, help_text='Enter a language for the book (e.g. English)')
+
+
+    class Meta:
+        ordering = ['language']
+
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return self.language
+
+
+class Book(models.Model):
+    """Model representing a book (but not a specific copy of a book)."""
+    title = models.CharField(max_length=200)
+
+    # Foreign Key used because book can only have one author, but authors can have multiple books
+    # Author as a string rather than object because it hasn't been declared yet in the file
+    author = models.ForeignKey('Author', on_delete=models.SET_NULL, null=True)
+
+    summary = models.TextField(max_length=1000, help_text='Enter a brief description of the book')
+    isbn = models.CharField('ISBN', max_length=13, unique=True,
+                             help_text='13 Character <a href="https://www.isbn-international.org/content/what-isbn">ISBN number</a>')
+
+    # ManyToManyField used because genre can contain many books. Books can cover many genres.
+    # Genre class has already been defined so we can specify the object above.
+    genre = models.ManyToManyField(Genre, help_text='Select a genre for this book')
+
+
+    # The language is stored as a OneToOneField because a book can have only one language
+    # The Language class is already defined, so it can be used to define the relation
+    language = models.ForeignKey('Language', Language, null = True)
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return self.title
+
+    def get_absolute_url(self):
+        """Returns the url to access a detail record for this book."""
+        return reverse('book-detail', args=[str(self.id)])
+
+
+
+class BookInstance(models.Model):
+    """Model representing a specific copy of a book (i.e. that can be borrowed from the library)."""
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, help_text='Unique ID for this particular book across whole library')
+    book = models.ForeignKey('Book', on_delete=models.RESTRICT, null=True)
+    imprint = models.CharField(max_length=200)
+    due_back = models.DateField(null=True, blank=True)
+
+    LOAN_STATUS = (
+        ('m', 'Maintenance'),
+        ('o', 'On loan'),
+        ('a', 'Available'),
+        ('r', 'Reserved'),
+    )
+
+    status = models.CharField(
+        max_length=1,
+        choices=LOAN_STATUS,
+        blank=True,
+        default='m',
+        help_text='Book availability',
+    )
+
+    class Meta:
+        ordering = ['due_back']
```

Comment:
Not sure but I think there might be a cause issue because in your model you set **due_back = models.DateField(null=True, blank=True)** ordering on null values data should not be format the correct result as expected.

Reply:
I understand what you mean by that and I tried to find solutions, but I am not sure how to fix this. It is possible for a book instance to not have a due_back date (if the book has not been borrowed, it wont have due back date). From what I understand, the null values are added to the end of list, so in this case, all borrowed books will come at the start of the list and the ones that are not borrowed will come at end (this is how I wanted to implement it). If the ordering would be like this, I think it would be fine to keep it as it is.